app/services/katello/event_daemon/runner.rb

```diff
@@ -87,7 +96,19 @@ def runnable?
 
         def register_service(name, klass)
           @services[name] = klass
+          @cache_keys[klass] = "#{name.downcase}_status"
         end
+
+        # writes status to cache if refresh is true or if there is no value cached
+        # otherwise, reads from cache
+        def cached_status(service_class:, status: nil, refresh: true)
+          cache_key = @cache_keys[service_class]
+          Rails.logger.debug "Writing status to cache #{cache_key}: #{status}" if refresh
+          Rails.cache.fetch(cache_key, force: refresh) do
+            status
+          end
```

Comment:
I went through several implementations and arrived at this one.  Originally this took a `cache_key` argument, but after I started removing the `STATUS_CACHE_KEY` constants from each service class I discovered that it's much simpler to just pass in the class here.  This is because `EventDaemon::Runner` is the only class that has access to the _names_ that the services were registered with, and the cache_key is derived from the name ( `"#{name.downcase}_status"`, i.e. `katello_events_status`.)